server/schedule/checker/merge_checker.go

```diff
@@ -134,8 +132,12 @@ func (m *MergeChecker) Check(region *core.RegionInfo) []*operator.Operator {
 }
 
 func (m *MergeChecker) checkTarget(region, adjacent *core.RegionInfo) bool {
+	if !m.cluster.IsGlobalMergeEnabled() {
```

Comment:
we need to check if `adjacent` is nil. I think it is better to abstracte it into a function `allowMerge` in `MergeChecker`.